cassandra/io/asyncorereactor.py

```diff
@@ -86,24 +88,22 @@ def maybe_start(self):
     def _run_loop(self):
         log.debug("Starting asyncore event loop")
         with self._loop_lock:
-            while True:
+            while not self._shutdown:
                 try:
-                    asyncore.loop(timeout=0.001, use_poll=True, count=1000)
+                    asyncore.loop(timeout=0.001, use_poll=True, count=100)
+                    self._timers.service_timeouts()
+                    if not asyncore.socket_map:
+                        time.sleep(0.005)
```

Comment:
I presume this is to avoid burning CPU while there are no conns?  Do you plan to switch to a non-global map instead of ``socket_map``?